api/api.proto

```diff
@@ -3229,6 +3238,73 @@ message SdkNodeInspectRequest {
   string node_id = 1;
 }
 
+// Options for draining volume attachment from a node
+message SdkNodeDrainAttachmentOptions {
+    // Selector is used for selecting volumes whose
+    // attachment needs to be moved from this node.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+    // Drain only sharedv4 volumes from a node
+    // By default all volumes will be drained.
+    bool only_sharedv4 = 2;
+}
+
+// Options for draining volume replicas from a node
+message SdkNodeDrainVolumeReplicaOptions {
+    // Selector is used for selecting volumes whose
+    // replicas need to be moved from this node to another.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+}
+
+// Options for draining/detaching cloud drives from a node
+message SdkNodeDrainCloudDriveOptions {
+    // When set the cloud drives will also get deleted.
+    bool delete_drives = 1;
+}
+
+// Options for draining kvdb from a node
+message SdkNodeDrainKvdbOptions {
+}
+
+
+// Options for draining a node in the cluster
+message SdkNodeDrainOptions {
+    // Perform all the node drain related operations.
+    // This is a single switch to perform all the supported
+    // node drain sub operations.
+    bool all = 1;
```

Comment:
if we want to use this as the default, we can remove it.

Because since it's a bool, if not supplied, it wills all=false, but our default implementation will perform all sub operations.

Reply:
I was thinking by default we will **not** do all the drain sub operations, but instead only do the operations which are set in the NodeDrainRequest. So this boolean is a single watch for callers of this API to perform all drain sub-operations with default settings.

Reply:
I have removed the boolean flag "all". The API is now more explicit, and the caller has to set what all sub operations need to be done as a part of this node drain request. The callers of this API whether its CLI or Operator should provide wrappers on performing sub operations.